config/config-server/resources/cruise-config.xsd

```diff
@@ -143,19 +143,58 @@
     <xsd:complexType name="configRepoType">
         <xsd:sequence>
             <xsd:choice minOccurs="1" maxOccurs="1">
-                <xsd:element name="svn" type="svnType"/>
-                <xsd:element name="hg" type="hgType"/>
-                <xsd:element name="p4" type="p4Type"/>
-                <xsd:element name="git" type="gitType"/>
-                <xsd:element name="tfs" type="tfsType"/>
-                <xsd:element name="scm" type="scmReferenceType"/>
+                <xsd:element name="svn" type="configRepoSvnType"/>
+                <xsd:element name="p4" type="configRepoP4Type"/>
+                <xsd:element name="git" type="configRepoGitType"/>
+                <xsd:element name="hg" type="configRepoHgType"/>
+                <xsd:element name="tfs" type="configRepoTfsType"/>
+                <xsd:element name="scm" type="configRepoScmReferenceType"/>
             </xsd:choice>
             <xsd:element name="configuration" minOccurs="0" maxOccurs="1" type="configurationType"/>
         </xsd:sequence>
         <xsd:attribute type="nameType" name="id" use="required"/>
         <xsd:attribute type="nameType" name="pluginId" use="optional"/>
     </xsd:complexType>
-
+    <xsd:complexType name="configRepoSvnType">
+        <xsd:attribute name="url" type="xsd:string" use="required"/>
+        <xsd:attribute name="username" type="xsd:string"/>
```

Comment:
`autoUpdate` has been removed from the configRepo material types. We need to allow users to be able to configure this for multiple reasons, one being that they might want to turn off autoUpdate and setup post-commit hooks for such materials. That should be allow. Another being if the fingerprint matches that of a material which is part of a pipeline, Go server wouldn't allow two different values for this flag and cause the same to fail.

Reply:
@jyotisingh -- I guess, `auto_update` for config repo materials is always set to true.
Not sure though why it should be always `true`.

https://github.com/gocd/gocd/blob/master/config/config-api/src/com/thoughtworks/go/config/remote/ConfigRepoConfig.java#L177-L182